src/library/scala/collection/ArrayOps.scala

```diff
@@ -125,11 +125,12 @@ object ArrayOps {
     private[this] val len = xs.length
     override def knownSize: Int = len - pos
     def hasNext: Boolean = pos < len
-    def next(): A = try {
+    def next(): A = {
+      if (pos >= xs.length) Iterator.empty.next()
```

Comment:
I had to use `xs.length` here instead of `len` not to break the JUnit test `scala.SerializationStabilityTest`. I have no idea why.

Reply:
Specialization 😞

`Stream.apply(1,2,3)` leads to serializing an `ArrayOps$ArrayIterator$mcI$sp`. Accessing `len` in the specialized method `next$mcI$sp` requires the `len` field in the base class `ArrayIterator` to be non-private. So it's name-mangled to `scala$collection$ArrayOps$ArrayIterator$$len`.

Reply:
Oh waw, that's deep! :astonished: Thanks for the explanation.